language/tools/vm-genesis/src/lib.rs

```diff
@@ -91,14 +90,20 @@ pub fn encode_genesis_transaction(
     )))
 }
 
-fn merge_txn_effects(
-    mut effects_1: TransactionEffects,
-    effects_2: TransactionEffects,
-) -> TransactionEffects {
-    effects_1.resources.extend(effects_2.resources);
-    effects_1.modules.extend(effects_2.modules);
-    effects_1.events.extend(effects_2.events);
-    effects_1
+fn squash_changesets(mut s1: MoveChangeSet, s2: MoveChangeSet) -> MoveChangeSet {
```

Comment:
These seems useful enough to potentially include in `impl MoveChangeSet` (you could imagine an optimization tha emits a writeset per-block instead of per-tx wanting it, for example).